### PR TITLE
Git Min Version Check

### PR DESCRIPTION
**Pull request by JimSuplizio**
The git commands used by the proxy server require us to be using a minimum version of git (2.37.0). For lab machines running automation this isn't a problem but for users using the CLI, this could be. 

- Added a function to GitProcessHandler, VerifyGitMinVersion, to verify that the minimum version of git is being used
- Added a GitVersionException which is thrown by VerifyGitMinVersion if the minimum version is not being used locally
- Added tests for VerifyGitMinVersion
- Added a call to VerifyGitMinVersion in Startup.cs, if the check fails the server exists with the appropriate message.
- Fixed the TestProxy's Common/Exceptions, HttpException and GitProcessException, which were in the wrong files with one being named incorrectly.
- Added a ToString override to the GitProcessException so it'll print out Result's CommandException. The reason for this is that ToString on the GitProcessException, gave no information as to what actually happened and we need that information, especially if something goes unhandled.
- Added code to the DebugLogger which writes using Console/Debug WriteLine functions in LogInformation and LogDebug if the logger isn't set. This was necessary because the DebugLogger isn't created unless the server is running and, in the case of using the CLI, it won't be. The async DebugLogger didn't need this because they're only used by the server when the server is running.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs [hunk 1]
```diff
@@ -60,7 +66,15 @@ public static bool CheckLogLevel(LogLevel level)
...
-            logger.LogInformation(details);
+            if (null != logger)
+            {
+                logger.LogInformation(details);
+            }
+            else
+            {
+                System.Console.WriteLine(details);
+            }
+
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs:77]: super nit: trailing newline.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs [hunk 2]
```diff
@@ -70,7 +84,15 @@ public static void LogInformation(string details)
...
-            logger.LogDebug(details);
+            if (null != logger)
+            {
+                logger.LogDebug(details);
+            }
+            else
+            {
+                Debug.WriteLine(details);
+            }
+
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs:95]: super nit: trailing newline.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/CommandResultException.cs [hunk 3]
```diff
@@ -1,46 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net;
-using System.Runtime.Serialization;
-using System.Threading.Tasks;
-
-namespace Azure.Sdk.Tools.TestProxy.Common.Exceptions
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/CommandResultException.cs:8]: Was this just a random file that I added erroneously and you're cleaning this up here?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/HttpException.cs [hunk 4]
```diff
@@ -4,17 +4,43 @@
 using System.Net;
 using System.Runtime.Serialization;
 using System.Threading.Tasks;
-using Azure.Sdk.Tools.TestProxy.Store;
 
 namespace Azure.Sdk.Tools.TestProxy.Common.Exceptions
 {
-    public class GitProcessException : Exception
+    public class HttpException : Exception
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/HttpException.cs:10]: I see. I somehow reversed the contents of these classfiles, and you're correcting this here. Thanks!

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs [hunk 5]
```diff
@@ -67,6 +67,10 @@ public static void Main(bool insecure = false, string storageLocation = null, st
                 Environment.Exit(0);
             }
 
+            GitProcessHandler gph = new GitProcessHandler();
+            // This throws and will exit
+            gph.VerifyGitMinVersion();
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs:72]:
```suggestion
            new GitProcessHandler().VerifyGitMinVersion();
```

**Review by scbedd — APPROVED** [review]
🚢🚢 🚢  thanks!

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/CommandResultException.cs (hunk 3 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/CommandResultException.cs:8]: @scbedd this wasn't a random file that I'm cleaning up. Here are the two exception classes
GitProcessException was in [HttpException.cs](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/HttpException.cs#L11)
HttpException was in [CommandResultException.cs](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/Exceptions/CommandResultException.cs#L10)
I wasn't cleaning, I was actually fixing these and putting them into their correct places.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs (hunk 2 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs:95]: Will be removed with next commit.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs (hunk 1 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/DebugLogger.cs:77]: Will be removed with next commit.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs (hunk 5 above)
Comment (JimSuplizio) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs:72]: Will be updated to the suggested change with the next commit.

**Pull request merged**